2007-03-29  Alan Modra  <amodra@bigpond.net.au>

	PR ld/4267
	* elf32-ppc.c (allocate_dynrelocs): Set plt.offset to -1 for
	unused entries.  Don't clear plt.plist in loop.

--- bfd/elf32-ppc.c	26 Mar 2007 12:23:00 -0000	1.211
+++ bfd/elf32-ppc.c	29 Mar 2007 01:11:30 -0000	1.212
@@ -4500,13 +4500,15 @@ allocate_dynrelocs (struct elf_link_hash
 	      }
 	    else
 	      ent->plt.offset = (bfd_vma) -1;
-
-	    if (!doneone)
-	      {
-		h->plt.plist = NULL;
-		h->needs_plt = 0;
-	      }
 	  }
+	else
+	  ent->plt.offset = (bfd_vma) -1;
+
+      if (!doneone)
+	{
+	  h->plt.plist = NULL;
+	  h->needs_plt = 0;
+	}
     }
   else
     {
